apexlog/apexlog.go

```diff
@@ -0,0 +1,143 @@
+package apexlog
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/airbrake/gobrake/v5"
+
+	"github.com/apex/log"
+)
+
+// severity identifies the sort of log: info, warning etc. It also implements
+// the flag.Value interface. The -stderrthreshold flag is of type severity and
+// should be modified only through the flag.Value interface. The values match
+// the corresponding constants in C++.
+type severity int8
+
+// These constants identify the log levels in order of increasing severity.
+// A message written to a high-severity log file is also written to each
+// lower-severity log file.
+const (
+	debugLog severity = iota
+	infoLog
+	warningLog
+	errorLog
+	fatalLog
+)
+
+var severityName = []string{
+	debugLog:   "debug",
+	infoLog:    "info",
+	warningLog: "warning",
+	errorLog:   "error",
+	fatalLog:   "fatal",
+}
+
+func severityByName(s string) (severity, bool) {
+	for i, name := range severityName {
+		if name == strings.ToLower(s) {
+			return severity(i), true
+		}
+	}
+	return 0, false
+}
+
+var gobrakeSeverity = errorLog
+
+// SetGobrakeSeverity sets minimum log severity that will be sent to Airbrake.
+//
+// Valid names are "debug" "info", "warning", "error", and "fatal".  If the name is not
```

Comment:
Why do we allow a situation when the name is not recognized? Instead of accepting `severity string` in `setGobrakeSeverity` we can accept type `severity`, therefore this:

```go
func setGobrakeSeverity(severity string) {
```

...becomes:

```go
func setGobrakeSeverity(s severity) {
```

Then we won't have to validate during runtime.